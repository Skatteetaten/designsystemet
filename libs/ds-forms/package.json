{
  "name": "@skatteetaten/ds-forms",
  "groupId": "no.skatteetaten.aurora",
  "version": "2.0.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^24",
    "react": "^19 ",
    "react-i18next": "^15",
    "@skatteetaten/ds-core-utils": "2.0.0",
    "@skatteetaten/ds-icons": "2.0.0",
    "@skatteetaten/ds-buttons": "2.0.0",
    "@skatteetaten/ds-typography": "2.0.0",
    "@skatteetaten/ds-progress": "2.0.0",
    "@skatteetaten/ds-status": "2.0.0",
    "date-fns": "^4",
    "@skatteetaten/ds-content": "2.0.0"
  },
  "scripts": {}
}
