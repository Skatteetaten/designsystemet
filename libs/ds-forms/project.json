{
  "name": "ds-forms",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-forms/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-forms",
        "tsConfig": "libs/ds-forms/tsconfig.lib.json",
        "project": "libs/ds-forms/package.json",
        "entryFile": "libs/ds-forms/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-forms/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-forms/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-forms"],
      "options": {
        "jestConfig": "libs/ds-forms/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-forms/**/*.css",
          "libs/ds-forms/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-forms"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-forms {args.switches}"
      }
    }
  }
}
