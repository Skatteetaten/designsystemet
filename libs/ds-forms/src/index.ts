export * from './Checkbox/Checkbox';
export * from './Checkbox/Checkbox.types';
export * from './CheckboxGroup/CheckboxGroup';
export * from './DatePicker/DatePicker.types';
export * from './DatePicker/DatePicker';
export * from './Select/Select';
export * from './Select/Select.types';
export * from './SelectOption/SelectOption.types';
export * from './CheckboxGroup/CheckboxGroup.types';
export * from './ErrorMessage/ErrorMessage';
export * from './ErrorMessage/ErrorMessage.types';
export * from './ErrorSummary/ErrorSummary';
export * from './ErrorSummary/ErrorSummary.types';
export * from './Fieldset/Fieldset';
export * from './Fieldset/Fieldset.types';
export * from './FileUploader/FileUploader';
export * from './FileUploader/FileUploader.types';
export * from './Radio/Radio';
export * from './Radio/Radio.types';
export * from './RadioGroup/RadioGroup';
export * from './RadioGroup/RadioGroup.types';
export * from './TextArea/TextArea.types';
export * from './TextArea/TextArea';
export * from './TextField/TextField.types';
export * from './TextField/TextField';
export * from './SearchField/SearchField';
export * from './SearchField/SearchField.types';
