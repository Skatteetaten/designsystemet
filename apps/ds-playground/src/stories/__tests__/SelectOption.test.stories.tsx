import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { Select } from '@skatteetaten/ds-forms';

const meta = {
  component: Select.Option,
  title: 'Tester/Select/Option',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    // HTML
    value: { table: { disable: true } },
  },
} satisfies Meta<typeof Select.Option>;
export default meta;
type Story = StoryObj<typeof meta>;

const Template: StoryFn<typeof Select.Option> = (args) => (
  <Select.Option {...args} />
);

const optionText = 'Option 1';
const optionValue = 'option1';
const defaultArgs = {
  children: optionText,
};

export const WithRef = {
  render: Template,
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLOptionElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const option = canvas.getByText(optionText);
    await expect(option).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  render: Template,
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const optionNode = canvas.getByText(optionText);
    await expect(optionNode).toHaveAttribute('id', 'htmlid');
    await expect(optionNode).toHaveAttribute('data-testid', '123ID');
    await expect(optionNode).toHaveClass('dummyClassname');
    await expect(optionNode).toHaveAttribute('lang', 'nb');
  },
} satisfies Story;

export const WithChildrenAndValue = {
  render: Template,
  name: 'With Children And Value',
  args: {
    ...defaultArgs,
    value: optionValue,
  },
  argTypes: {
    children: { table: { disable: false } },
    value: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const optionNode = canvas.getByText(optionText);
    await expect(optionNode).toBeInTheDocument();
    await expect(optionNode).toHaveValue(optionValue);
  },
} satisfies Story;
