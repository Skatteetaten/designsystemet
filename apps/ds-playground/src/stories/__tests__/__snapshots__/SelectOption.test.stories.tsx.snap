// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Select/Option WithAttributes play-test 1`] = `
<div data-test-block="true">
  <option lang="nb"
          id="htmlid"
          class="_option_idcf7_1 dummyClassname"
          data-testid="123ID"
  >
    Option 1
  </option>
</div>
`;

exports[`Tester/Select/Option WithChildrenAndValue play-test 1`] = `
<div data-test-block="true">
  <option class="_option_idcf7_1"
          value="option1"
  >
    Option 1
  </option>
</div>
`;

exports[`Tester/Select/Option WithRef play-test 1`] = `
<div data-test-block="true">
  <option class="_option_idcf7_1"
          id="dummyIdForwardedFromRef"
  >
    Option 1
  </option>
</div>
`;
